doc/design/dist/README.md

```diff
@@ -0,0 +1,190 @@
+# Distributed Training Design Doc
+
+## Objective
+
+We want Paddle to support training on the general-purpose cluster. The cluster runs Paddle, the web server (e.g., Nginx), the log collector (e.g., fluentd), the distributed queue service (e.g., Kafka), the log joiner and other data processors written using Storm, Spark, and Hadoop MapReduce on the same cluster. As illustrated in the following graph:
+
+<img src="src/arch.png"/>
+
+This poses new challenges for Paddle,
+
+- Paddle need to be fault tolerant.
+- Input training data can be online data from real time logs or batch data from distributed file system.
+- User needs a simple way to train model on Paddle cloud. Complexities such as job scheduling should be hidden from user.
+
+## Training Job
+
+A training job will be created once user asks Paddle cloud to train a model. The training job is made up of different processes that collaboratively consume data and produce a trained model. There are three kinds of processes:
+
+- Master process
```

Comment:
To describe the roles in a concise but clear way, we might write:

-----

1. the master process, which dispatches tasks to 
1. one or more trainer processes, which run distributed training and synchronize gradients/models via
1. one or more parameter server processes, where each holds a shard of the global model.